exposee-ui/src/components/Profilepage/Profile.jsx

```diff
@@ -1,46 +1,78 @@
-import "./Profile.css";
-import { useState, useEffect, useContext } from "react";
-import { UserContext } from "../../UserContext"; 
-import {Link} from "react-router-dom";
-import axios from "axios";
+import React, { useState, useEffect, useContext } from "react";
+import { UserContext}  from '../../UserContext';
+import { Link, useNavigate } from "react-router-dom";
+//import axios from "axios";
 
-const Profilepage = ()=> {
-    const {user, setUser} = useContext(UserContext);
-    useEffect (() => {
-        const fetchUserProfile = async ()=>{
-        try {
-            const response = await fetch(`http://localhost:3000/profile/${user.id}`);
-        }catch(error){
-            console.error('Error fetching user profile: ', error);
+const ProfilePage = () => {
+  const { user, setUser } = useContext(UserContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const navigate = useNavigate();
+  useEffect(() => {
+    const fetchUserProfile = async () => {
+      try {
+        const response = await fetch(`http://localhost:3000/profile/${user.id}`);
+        if (response.ok) {
+          const data = await response.json();
+          setUser(data);
+        } else {
+          console.log("Failed to fetch user profile");
         }
+        setIsLoading(false);
+      } catch (error) {
+        console.error('Error fetching user profile:', error);
+        setIsLoading(false);
+      }
     };
+
     fetchUserProfile();
-    }, []);
-    const handleBroadcast = async () => {
-        try {
-            const response = await fetch ('/broadcast');
-            console.log(Response.data);
-        }catch (error){
-            console.error('Error going live: ', error);
-        }
-    }; console.log(user.profilePicture)
-    if (!user){
-        return <div>Loading...</div>;
+  }, [user.id, setUser]);
+console.log(user.access_token);
+
+  const handleBroadcast = async () => {
+    try {
+      const response = await fetch('http://localhost:3000/broadcast', {
+        method: "POST",
+        headers: {
+            'Content-Type': 'application/json',
+            access_token: user.access_token
+        },
+        body: JSON.stringify({
+          description: "STREAM",
+        }),
+      });
+console.log('helloe',response);
+      if (response.ok) {
+        const data = await response.json();
+        alert(`Broadcast started successfully!, use the pop up stream key to set up stream in OBS . STREAMKEY: ${data.mux_stream_key
+        }`);
+        console.log(data);
+      } else {
+        console.log("Failed to start the live stream");
+      }
+    } catch (error) {
+      console.error('Error going live:', error);
     }
-    return (
+  };
+
+  if (isLoading) {
```

Comment:
lol it seems you already did a loading spinner for the profile page??

Reply:
i thought u wanted it to be in the video
.. I told me if i am feeling fancy i can do it on the video too